src/js/components/sharedComponents/VerticalLine.jsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * VerticalTodayLine.jsx
+ * Created by Jonathan Hill 07/05/19
+ **/
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { throttle } from 'lodash';
+
+const propTypes = {
+    xScale: PropTypes.func, // function to set line position
+    text: PropTypes.string, // text to display with line
+    y1: PropTypes.number, // top of graph
+    y2: PropTypes.number, // bottom of graph
+    xMax: PropTypes.number, // max possible value of x
+    xMin: PropTypes.number, // max possible value of x
+    xValue: PropTypes.number, // actual value of x
+    showTextRight: PropTypes.bool, // show text to the right of line
+    showTextLeft: PropTypes.bool, // show text to the left of line
+    textY: PropTypes.number, // show text at this height
+    description: PropTypes.string
+};
+
+export default class VerticalLine extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            textX: 0,
+            textY: props.textY,
+            windowWidth: null
+        };
+        this.handleWindowResize = throttle(this.handleWindowResize.bind(this), 50);
+        this.textDiv = null;
+        // React will call this function when the DOM draws it ( React Callback Refs )
+        this.setTextDiv = (element) => {
+            this.textDiv = element;
+            this.positionText();
+        };
+    }
+
+    componentDidMount() {
+        this.handleWindowResize();
+        window.addEventListener('resize', this.handleWindowResize);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.handleWindowResize);
+    }
+    // since we set the position of the text we need to update it on window resize
+    handleWindowResize() {
+        const windowWidth = window.innerWidth;
+        if (this.state.windowWidth !== windowWidth) {
+            this.setState({ windowWidth });
+            this.positionText();
+        }
+    }
+
+    description() {
+        return this.props.description || 'A vertical line representing today\'s date';
```

Comment:
This could just be a const defined by a ternary, or just defined with the `||` operator.

Reply:
Done in other PR